core/pkg/ingress/controller/annotations.go

```diff
@@ -0,0 +1,116 @@
+/*
+Copyright 2017 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package controller
+
+import (
+	"github.com/golang/glog"
+
+	"k8s.io/kubernetes/pkg/api"
+	"k8s.io/kubernetes/pkg/apis/extensions"
+
+	"k8s.io/ingress/core/pkg/ingress/annotations/auth"
+	"k8s.io/ingress/core/pkg/ingress/annotations/authreq"
+	"k8s.io/ingress/core/pkg/ingress/annotations/authtls"
+	"k8s.io/ingress/core/pkg/ingress/annotations/cors"
+	"k8s.io/ingress/core/pkg/ingress/annotations/healthcheck"
+	"k8s.io/ingress/core/pkg/ingress/annotations/ipwhitelist"
+	"k8s.io/ingress/core/pkg/ingress/annotations/parser"
+	"k8s.io/ingress/core/pkg/ingress/annotations/proxy"
+	"k8s.io/ingress/core/pkg/ingress/annotations/ratelimit"
+	"k8s.io/ingress/core/pkg/ingress/annotations/rewrite"
+	"k8s.io/ingress/core/pkg/ingress/annotations/secureupstream"
+	"k8s.io/ingress/core/pkg/ingress/annotations/sslpassthrough"
+	"k8s.io/ingress/core/pkg/ingress/defaults"
+	"k8s.io/ingress/core/pkg/ingress/errors"
+)
+
+type extractorConfig struct {
+	backdefs           defaults.Backend
+	getSecret          func(string) (*api.Secret, error)
+	getAuthCertificate func(string) (*authtls.SSLCert, error)
+}
+
+type annotationExtractor struct {
+	annotations map[string]parser.IngressAnnotation
+}
+
+func newAnnotationExtractor(cfg extractorConfig) annotationExtractor {
+	return annotationExtractor{
+		map[string]parser.IngressAnnotation{
+			"BasicDigestAuth": auth.NewParser(cfg.getSecret),
+			"ExternalAuth":    authreq.NewParser(),
+			"CertificateAuth": authtls.NewParser(cfg.getAuthCertificate),
+			"EnableCORS":      cors.NewParser(),
+			"HealthCheck":     healthcheck.NewParser(cfg.backdefs),
+			"Whitelist":       ipwhitelist.NewParser(cfg.backdefs),
+			"Proxy":           proxy.NewParser(cfg.backdefs),
+			"RateLimit":       ratelimit.NewParser(),
+			"Redirect":        rewrite.NewParser(cfg.backdefs),
+			"SecureUpstream":  secureupstream.NewParser(),
+			"SSLPassthrough":  sslpassthrough.NewParser(),
+		},
+	}
+}
+
+func (e *annotationExtractor) Extract(ing *extensions.Ingress) map[string]interface{} {
+	anns := make(map[string]interface{}, 0)
+	for name, fn := range e.annotations {
```

Comment:
`fn` isn't a `func`, so this naming is confusing.

Maybe `name, annotationParser`?